onnxruntime/server/executor.cc

```diff
@@ -36,12 +37,10 @@ protobufutil::Status Executor::SetMLValue(const onnx::TensorProto& input_tensor,
     return GenerateProtobufStatus(status, "GetSizeInBytesFromTensorProto() failed: " + status.ToString());
   }
 
-  std::unique_ptr<uint8_t[]> data(new uint8_t[cpu_tensor_length]);
-  memset(data.get(), 0, cpu_tensor_length);
-
   OrtCallback deleter;
+  auto* buf = buffers.AllocNewBuffer(cpu_tensor_length);
   status = onnxruntime::utils::TensorProtoToMLValue(onnxruntime::Env::Default(), nullptr, input_tensor,
```

Comment:
Please don't rely on the 'onnxruntime::utils::TensorProtoToMLValue' function. It's an internal function.

Reply:
onnx server's usage of ORT is some ways similar to WinML. It doesn't use the C API; they use the internal C++ API and link the libs statically.

Reply:
Is there any concern about using internal APIs here? Any plan to change the implementation in the near future? @snnn

Reply:
Any resolution here? I can write our own `TensorProtoToMLValue` if need be

Reply:
Whether it's WimML or onnx server, the interface between it and ORT should be simple and clean.

Reply:
I would suggest any code in this repo and use static link could use internal C++ APIs or C APIs. Any product out of this repo or using dynamic link should only use C APIs.

Reply:
OK then I propose we do this as part of a larger task to move this project to use the [C++ API](https://github.com/microsoft/onnxruntime/blob/master/include/onnxruntime/core/session/onnxruntime_cxx_api.h). Otherwise we would be duplicating a lot of efforts (like this one for example).

Thoughts?

Reply:
> I would suggest any code in this repo and use static link could use internal C++ APIs or C APIs
Unless every dev in this team knows how to debug onnx server's code.

Reply:
Earlier we explicitly made the decision that onnx server will use the internal C++ APIs. Then why is the dependence on core/framework/tensorprotoutils.h worse than core/session/inference_session.h? The internal C++ APIs are also subject to change just like tensorprotoutils.h, right? I would prefer that we don't duplicate code and introduce bugs as part of it.

Reply:
> Unless every dev in this team knows how to debug onnx server's code.

@snnn I totally get your concern. As long as we do not need dup the code, I'm happy to have a backlog item to move to a formal C++ API when it is ready. But I would prefer to scope this PR only the bug fix and the unit test. 

In the meanwhile. if anyone in ONNX Runtime team need help on debugging the server code, Trent and I will be more than happy to help.

Reply:
I revoke my word. But, in any case, software should be built in layered. And there should be a clear interface between layers. onnx server is an application server, it shouldn't talk to either HTTP internals or ORT inference engine internals. Even in some case we need to duplicate a few code, it is still worth to. 

In this case, TensorProtoToMLValue is highly tight to session state initializer.  And I'm going to change its interface because the current implementation is not optimal, because session state initializer doesn't know how TensorProtoToMLValue would allocate the memory, but session state initializer need to the details and take the ownership of the buffers.